src/lib/services-manager/service_loader.rb

```diff
@@ -0,0 +1,194 @@
+    def read
+      @services   = {}
+      @unit_files = {}
+      @units      = {}
+
+      load_unit_files
+      load_units
+
+      @supported_unit_files = unit_files.select do |_, status|
+        status != Status::MASKED # masked services should not been shown at all
+      end
+
+      @supported_units = units.reject do |name, attributes|
+        attributes[:status] == Status::NOTFOUND # definition file is not available anymore
+      end
+
+      extract_services
+      services
+    end
+
+  private
+
+    # FIXME: use Yast::Systemctl for this, remember to chomp SERVICE_SUFFIX
+
+    # @return [Array<String>] "apache2.service   enabled\n"
+    def list_unit_files
+      command = TERM_OPTIONS + LIST_UNIT_FILES_COMMAND + COMMAND_OPTIONS
+      out = Yast::SCR.Execute(Yast::Path.new('.target.bash_output'), command)['stdout']
+      out.lines
+    end
+
+    # @return [Array<String>] "dbus.service   loaded active running D-Bus System Message Bus\n"
+    def list_units
+      command = TERM_OPTIONS + LIST_UNITS_COMMAND + COMMAND_OPTIONS
+      out = Yast::SCR.Execute(Yast::Path.new('.target.bash_output'), command)['stdout']
+      out.lines
+    end
+
+    def load_unit_files
+      list_unit_files.each do |line|
+        service, status = line.split(/[\s]+/)
+        service.chomp! SERVICE_SUFFIX
+        # Unit template, errors out when inquired with `systemctl show`
+        # See systemd.unit(5)
+        next if service.end_with?("@")
+        unit_files[service] = status
+      end
+    end
+
+    def load_units
+      list_units.each do |line|
+        service, status, _active, _sub_state, *description = line.split(/[\s]+/)
+        service.chomp! SERVICE_SUFFIX
+        units[service] = {
+          :status => status,
+          :description => description.join(' ')
+        }
+      end
+    end
+
+    def extract_services_from_unit_files
+      @supported_unit_files.each do |name, status|
+        services[name] = DEFAULT_SERVICE_SETTINGS.clone
+        if @supported_units[name]
+          # Services are loaded into the system. Taking that one because there are more
+          # information
+          services[name][:loaded] = @supported_units[name][:status] == Status::LOADED
+          services[name][:description] = @supported_units[name][:description]
+        end
+        services[name][:can_be_enabled] = status == Status::STATIC ? false : true
+      end
+    end
+
+    def extract_services_from_units
+      @supported_units.each do |name, service|
+        next if services[name]
+        services[name] = DEFAULT_SERVICE_SETTINGS.clone
+        services[name][:loaded] = service[:status] == Status::LOADED
+        services[name][:description] = service[:description]
+      end
+    end
+
+    def extract_services
+      extract_services_from_unit_files
+      # Add old LSB services (Services which are loaded but not available as a unit file)
```

Comment:
is it still available on system? I think it no longer exists, but it should be verified.